Subject: Fix test compatibility with pytest >= 9.1

Pytest 9.1 no longer accepts non-Collection iterables in parametrize.
Convert the generator expression to a list comprehension.

Backported from https://github.com/dateutil/dateutil/pull/1523

diff --git a/tests/test_isoparser.py b/tests/test_isoparser.py
--- a/tests/test_isoparser.py
+++ b/tests/test_isoparser.py
@@ -120,8 +120,8 @@
 DATETIMES = [datetime(2017, 11, 27, 6, 14, 30, 123456)]
 @pytest.mark.parametrize('dt', tuple(DATETIMES))
 @pytest.mark.parametrize('date_fmt', YMD_FMTS)
-@pytest.mark.parametrize('time_fmt', (x + sep + '%f' for x in HMS_FMTS
-                                      for sep in '.,'))
+@pytest.mark.parametrize('time_fmt', [x + sep + '%f' for x in HMS_FMTS
+                                      for sep in '.,'])
 @pytest.mark.parametrize('tzoffset', TZOFFSETS)
 @pytest.mark.parametrize('precision', list(range(3, 7)))
 def test_ymd_hms_micro(dt, date_fmt, time_fmt, tzoffset, precision):
